### DISCUSSION TITLE
Welcome to SX126x-Arduino Discussions!

### DISCUSSION THREAD

**Discussion by beegee-tokyo** [beegee-tokyo/SX126x-Arduino#112] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zarie3** [comment]
Hi,
I am trying to use the library with ESP32C3 and SX1262 chip, in the HT-CT62 module in a LoRaWan comm.
The LoRaWan example builds and uploads properly however, when the ESP32C3 boots up it resets every time, I think it has to do with the SPI.
Does the library was tested also on ESP32C3 ?
Thank you

**Reply by joyel24** [reply]
> Hi, I am trying to use the library with ESP32C3 and SX1262 chip, in the HT-CT62 module in a LoRaWan comm. The LoRaWan example builds and uploads properly however, when the ESP32C3 boots up it resets every time, I think it has to do with the SPI. Does the library was tested also on ESP32C3 ? Thank you

hello @zarie3,
It's a late reply but I recently ordered a lot of this HT-CT62 modules. They will be in my mailbox around 07/15 so I did nothing yet.
I plan to use them as repeaters for meshtastic.

I own some ESP32-C3-WROOM-02, it's very closed to the HT-CT62 but without SX1262 of course.
If I power them with 3.3V they resets themself almost everytime, especially when using wifi.
It's simply solved by adding a capacitor.

I hope your trouble can be resolved by adding a capacitor in parallel to the power. Closest as you can to the chip for more stability.

I think you found a solution but my answer can be useful for other people.

**Reply by zarie3** [reply]
Hi Joyel,
Thank you for your sincere reply !.
I reverted back to Heltec's library, where I am getting tech support, as we use the module for production purposes.
However I also thought on using the module with meshtastic for other usages, it shall take us time after we conclude the current project implementation.
I shall check the schematics and propose the addition of such capacitor close to the 3.3 supply pin.

**Reply by joyel24** [reply]
Talking aboout HT-CT62, Today I found this opensource board made for it by paulhausk, there was an issue with USB and I think I fixed it but can't confirm yet.
The fixed board is here :
https://gitlab.com/joyel24/CheapMesh

**Comment by beegee-tokyo** [comment]
I never tested it on a EPS32-C3, I don't have one. Only Wrover and Wroom versions of the ESP32 were tested.

**Comment by DavidVanHorn** [comment]
I'm having issues using the library.  There seem to be dependencies on libraries which are not supplied. 
I'm looking to compile for the [Heltec V3](https://heltec.org/project/wifi-lora-32-v3/)

Radio.h, nrf_gpio.h   Could be others.
How can I resolve this?

What I'm trying to get to is a very simple program to use the Heltek module as a signal source for antenna pattern testing.
The only thing I need the radio to do is raise an unmodulated carrier, and go back to receive. 
The functions   SX126xSetTxContinuousWave();  and   SX126xSetRx (); should do the trick, but I'd like to at least get the demo app to compile and run before I start hacking on it. :)

**Comment by beegee-tokyo** [comment]
I heard about issues with the Heltec boards before, but I don't have one and I am not sure why it throws errors.
It works fine and all required files are in the library when compiling for nRF52, ESP32 or RP2040.

Heltec has their own [LoRa library](https://github.com/HelTecAutomation/Heltec_ESP32), maybe you can try theirs.

**Reply by DavidVanHorn** [reply]
I downloaded your zip, uncompressed it, and tried to compile the examples.   Did I miss something?

**Reply by beegee-tokyo** [reply]
As I said, Heltec has their own LoRa library and from the board description, their library is included in their BSP:
![image](https://github.com/user-attachments/assets/65d419de-35ec-415a-81fb-f156e57bbc59)

You are getting these errors most likely because of conflicts between their LoRaWAN library and mine.

I guess you have to use their LoRaWAN library instead.

**Comment by DavidVanHorn** [comment]
Ok, I uninstalled the Arduino environment, scrubbed all the files from my system, did a fresh install of the Arduino IDE, and still hit the same problem.    I'm not that experienced with Arduino, so I'm betting I have something misconfigured, but IDK what it would be.  I'm more experienced with AVR Studio, MPLab, and Eclipse.

**Comment by beegee-tokyo** [comment]
What BSP are you using to compile?
If you are still using the Heltec one, you will have the same problem.
Uninstall my library and check with Heltec documentation how to use their integrated LoRaWAN library. I am sure they have examples.

**Comment by taha633** [comment]
Hi everyone,
I'm having a problem with wiring the ESP32 to the E22.
I read the ReadMe, but I didn't understand what is meant by "connect all the E22 pins to the ESP32 pins"! Does it mean ESP32 GPIO pins or physical pins?
